src/ZkFold/Cardano/Scripts/ForwardingScripts.hs

```diff
@@ -32,3 +33,24 @@ forwardingMint _label symbolHash _ ctx =
 
         -- Finding scripts to be executed
         reds = keys $ txInfoRedeemers $ scriptContextTxInfo ctx
+
+untypedForwardingReward :: BuiltinData -> BuiltinData -> BuiltinData -> BuiltinUnit
```

Comment:
I suggest adding INLINABLE pragma, that is to say `{-# INLINABLE untypedForwardingReward #-}` .

Also, type signature for `untypedForwardingReward` should be `BuiltinData -> BuiltinUnit`.  Suggested chagne:
```haskell
untypedForwardingReward :: BuiltinData -> BuiltinUnit
untypedForwardingReward ctx' =
  let ctx = unsafeFromBuiltinData ctx' in
    case scriptContextScriptInfo ctx of
      SpendingScript _ (Just dat) ->
        let datum    = unsafeFromBuiltinData . getDatum $ dat
            redeemer = unsafeFromBuiltinData . getRedeemer . scriptContextRedeemer $ ctx
        in
          check $
            forwardingReward
              datum
              redeemer
              ctx
      _                           -> error ()
```